### DISCUSSION TITLE
Disable Triangulation cannot be accessed in IFCGeomSettings

### DISCUSSION THREAD

**Discussion by DprOliv** [IfcOpenShell/IfcOpenShell#5652] [General]
Hello all

Trying to use here ifcopenShell.geom iterators ...
Everything is running well excpet to the fact that i cannot use DISABLE TRIANGULATION inside ifcGeom.settings() ... 
As follows...

import ifcopenshell.geom as ifcGeom
tree_settings = ifcGeom.settings()
tree_settings.set(tree_settings.DISABLE_TRIANGULATION, True)

When  try to do this i get the error 
"'Settings' object has no attribute 'DISABLE_TRIANGULATION' 

Ay hints ?

Thanks !

**Comment by aothms** [comment]
```
ifcopenshell.geom.settings(ITERATOR_OUTPUT=ifcopenshell.ifcopenshell_wrapper.NATIVE)
```

**Reply by DprOliv** [reply]
@aothms thank you for your hint.
However i still cannot set DISABLE_TRIANGULATION attribute ... getting the same error as i mentioned
A related question: Without setting DISABLE_TRIANGULATION to True i cannot use ifcGeom.tree iterator ... is this valid ?

**Reply by aothms** [reply]
DISABLE_TRIANGULATION  is replaced by ITERATOR_OUTPUT.

ifcgeom.tree indeed requires the DISABLE_TRIANGULATION / ITERATOR_OUTPUT=NATIVE

**Reply by DprOliv** [reply]
Got it now ... i was (mistakenly...) understanding that this was a set to make geom.settings to process the DISABLE TRIANGULATION ... 

Tree usage is working perfectly now here !
Thank you for your clarification and guidance @aothms  !

As one last (related ...) question: as for Geometry processing which is the best source of information (considering i'm using Python wrapper ... of IFCOS ...) ? 
This one: https://docs.ifcopenshell.org/autoapi/ifcopenshell/geom/index.html
?
I understand that there are also the util.shape functions (several useful functions for many geometry related processing...), right ?

**Reply by aothms** [reply]
>  I understand that there are also the util.shape functions (several useful functions for many geometry related processing...), right ?

Yes, and if you really want to do detailed things there is probably no way around also using pythonocc for things. So it's hard to make a recommendation there. Improvements to the docs are always very welcome though. If you spot outdated things or see a place for a higher level introduction, pull requests are always welcome and the community can refine and fill in things over time.

**Reply by DprOliv** [reply]
Ok. Got it. Thanks @aothms.
 I remember using Python OCC sometime ago..... and that (at that time, at least...) it was very challenging to install PythonOCC.
